proposals/gc/Overview.md

```diff
@@ -297,20 +300,19 @@ Elements are accessed with generic load/store instructions that take a reference
 )
 ```
 The element type of every access is checked at validation time.
-The index is checked at execution time.
+The index is checked against the array's length at execution time.
 A trap occurs if the index is out of bounds.
 
-Arrays are *allocated* with a `new` instruction that takes a size and an initialization value as operands, yielding a reference:
+Arrays are *allocated* with a `new` instruction that takes a length and an initialization value as operands, yielding a reference:
 ```
 (func $g
   (call $g (new_array $vector (i32.const 0) (i64.const 3.14)))
 )
 ```
-Arrays are garbage-collected.
 
-The *length* of an array, i.e., the number of elements, can be inquired via the `load_length` instruction:
+The *length* of an array, i.e., the number of elements, can be inquired via the `load_array_length` instruction:
 ```
-(load_length $vector (get_local $v))
+(load_array_length $vector (get_local $v))
```

Comment:
This is to make it explicit, since other structures might have lengths in the future. "array_length" is a thing in the "verb_thing" sense, not a "verb_type_thing" sense.